java/arcs/core/allocator/Arc.kt

```diff
@@ -119,23 +122,25 @@ class Arc internal constructor(
         }
     }
 
-    private fun registerChangeHandlerWithArcHosts() = sync(this) {
+    private suspend fun registerChangeHandlerWithArcHosts() {
         require(registration == null) {
             "registration called more than once"
         }
-        runBlocking {
-            // first poll the current states of all hosts
-            fetchCurrentStates()
+        // first poll the current states of all hosts
+        fetchCurrentStates()
 
-            // Register event listeners
-            partitions.forEach { partition ->
-                val arcHost = allocator.lookupArcHost(partition.arcHost)
-                registration = arcHost.addOnArcStateChange(id) { _, state ->
-                    sync(this) {
+        // Register event listeners
+        partitions.forEach { partition ->
+            val arcHost = allocator.lookupArcHost(partition.arcHost)
+            registration = arcHost.addOnArcStateChange(id) { _, state ->
```

Comment:
Why do we re-set an instance variable for each partition? Is it okay that we lose the values for all but the `n`th arcHost?

Reply:
its bug, but unregistering and heartbeats are not implemented yet, so no consequences, but I fixed it, thanks for catching it.